pulsar-broker/src/test/java/org/apache/pulsar/client/api/MutualAuthenticationTest.java

```
@@ -0,0 +1,225 @@
+                return new MutualAuthenticationDataProvider();
+            } catch (Exception e) {
+                throw new PulsarClientException(e);
+            }
+        }
+
+        @Override
+        public void configure(Map<String, String> authParams) {
+            // noop
+        }
+
+        @Override
+        public void start() throws PulsarClientException {
+            // noop
+        }
+    }
+
+
+    public static class MutualAuthenticationState implements AuthenticationState {
+        @Override
+        public String getAuthRole() throws AuthenticationException {
+            return "admin";
+        }
+
+        @Override
+        public AuthData authenticate(AuthData authData) throws IOException {
+            String clientData = new String(authData.getBytes(), Charset.forName("UTF-8"));
+            assertEquals(clientData, clientAuthStrings[authStep]);
+            log.debug("authenticate in server. passed in :{}, send: serverAuthStrings[{}]: {}",
+                clientData, authStep, serverAuthStrings[authStep]);
+            AuthData serverData =  AuthData.of(serverAuthStrings[authStep ++].getBytes());
+            return serverData;
+        }
+
+        @Override
+        public AuthenticationDataSource getAuthDataSource() {
+            return null;
+        }
+
+        @Override
+        public boolean isComplete() {
+            return authStep > 1;
+        }
+    }
+
+    public static class MutualAuthenticationProvider implements AuthenticationProvider {
+        @Override
+        public void close() throws IOException {
+        }
+
+        @Override
+        public void initialize(ServiceConfiguration config) throws IOException {
+        }
+
+        @Override
+        public String getAuthMethodName() {
+            return "MutualAuthentication";
+        }
+
+        @Override
+        public String authenticate(AuthenticationDataSource authData) throws AuthenticationException {
+            return "admin";
```

should not be used so should throw an exception.

---

oh, test and recall that, this is also used in web service(like admin). try to change that part after this pr.